packages/utilities-react/src/StepFlow/StepGroup.tsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect } from 'react';
+import { useStepContext } from '.';
+
+interface StepGroupProps {
+  children: React.ReactNode;
+}
+
+/**
+ * Container for steps, each child represents a step
+ */
+export const StepGroup = ({ children }: StepGroupProps) => {
+  const { setTotalSteps, currentStep } = useStepContext();
+
+  const childrenArray = React.Children.toArray(children);
+  const cleanedChildren = childrenArray.filter(Boolean);
+  const childrenCount = React.Children.count(cleanedChildren);
+
+  // set total step count
+  useEffect(() => {
+    setTotalSteps(childrenCount);
+  }, [childrenCount, setTotalSteps]);
+
+  const currentStepComponent =
+    React.Children.toArray(cleanedChildren)[currentStep - 1];
+
+  // return only the current step
+  return currentStepComponent;
+};
```

Comment:
I see `React.Children.toArray(cleanedChildren)` but isn't `cleanedChildren` already that?

Also, the code comment isn't the most accurate since you're filtering out falsy children. '[E]ach child' won't always represent a step.

```suggestion
export const StepGroup = ({ children }: StepGroupProps) => {
  const { setTotalSteps, currentStep } = useStepContext();

  const validChildren = React.Children.toArray(children).filter(Boolean);

  useEffect(() => {
    setTotalSteps(validChildren.length);
  }, [validChildren.length, setTotalSteps]);

  return validChildren[currentStep - 1];
};
```

Reply:
Good point, I see now that `Children.toArray` doesn't return `null` or `undefined` values. I'll update that as well as a more accurate comment.

Reply:
Addressed in https://github.com/carbon-design-system/carbon/pull/20389/commits/77aaf12e3020cc62c39c475d6cd8dc538f040f67